core/domain/state_domain.py

```diff
@@ -540,6 +577,70 @@ def from_dict(cls, interaction_dict):
             [Hint.from_dict(h) for h in interaction_dict['hints']],
             solution_dict)
 
+    def to_proto(self):
+        """Returns a proto representation of the interaction object.
+
+        Returns:
+            InteractionInstance|dict. The InteractionInstance proto object
+            if the id of the interaction matches else return an empty dict.
+        """
+        interaction_instance = None
+        interaction_proto = {}
+        if self.id is not None:
+            interaction_instance = (
+                interaction_registry.Registry.get_interaction_by_id(self.id)
+            )
+
+        if self.id == 'Continue':
+            interaction_proto = state_pb2.InteractionInstance(
+                continue_instance=interaction_instance.to_proto(
+                    self.default_outcome, self.customization_args))
+        elif self.id == 'FractionInput':
```

Comment:
How can we convert this if elif approach in a way where we can only use `android_validation_constants.VALID_INTERACTION_IDS` and make this interaction proto creating work? 

The only hurdle here is every interaction `to_proto` function receives arguments differently, for example:
1. EndExploration to_proto function needs no argument.
2. Continue to_proto function needs no solution/answer group/hints. 
We cannot pass self as a whole, which will solve the issue but not a good approach as we are not inclined towards passing self as argument. 

The other thing is the proto id name is different for each interaction:
```
state_pb2.InteractionInstance(
  continue_instance=...)

state_pb2.InteractionInstance(
  fraction_input=...)

state_pb2.InteractionInstance(
  item_selection_input=...)
.
.
.
```

This might get solved by creating a map of proto id and interaction name like, but not sure with this approach here:
```py
interaction_proto_id_mapping = {
    'Continue': 'continue_instance',
    'FractionInput': 'fraction_input',
    'ItemSelectionInput': 'item_selection_input',
    'MultipleChoiceInput': 'multiple_choice_input',
    'NumericInput': 'numeric_input',
    'TextInput': 'text_input',
    'RatioExpressionInput': 'ratio_expression_input',
    'ImageClickInput': 'image_click_input',
    'DragAndDropSortInput': 'drag_and_drop_sort_input',
    'EndExploration': 'end_exploration'
}
```

Reply:
Okay, probably fine to keep the elif approach.